### ISSUE TITLE
Deploy consul as a stateful set as alternative to etcd

### ISSUE THREAD

**Issue by dbason** [banzaicloud/bank-vaults#961]
**Is your feature request related to a problem? Please describe.**
The etcd operator is basically abandoned and isn't very resilient.  Most of the issues we have had with the bank-vaults operator have been down to etcd under the hood.

**Describe the solution you'd like**
The operator should optionally deploy a conul statefulset as an alternative to using the etcd operator to deploy and etcd cluster.

**Describe alternatives you've considered**
We could manually deploy the consul helm charts, but I think one of the benefits of the etcd set up is that it's integrated with the operator and can set up all of the tls secrets.  It would be nice to do the same thing with a consul backend.

**Comment by bonifaido** [comment]
Yeah, I agree, the etcd-operator is archived, we should probably deprecate the feature entirely of configuring etcd via the operator.

We suggest users to use the Raft backend, it became stable in Vault 1.4, and offers HA storage without an external entity like Consul in this case. Wouldn't that work for you @dbason ?

**Comment by dbason** [comment]
I wasn't aware the raft storage had gone stable.  That being the case I think that will be perfect.

**Comment by bonifaido** [comment]
We are going to check Vault 1.4 in the coming days, I think Bank-Vaults should work fine with it, but we need to test it.

**Comment by bonifaido** [comment]
Anyhow, I'm not against Consul as a supported backend.
